File: javascript/src/Creational/Prototype/PrototypeFactory.js
(function() {
  define([], function() {
    var PrototypeFactory;
    PrototypeFactory = (function() {
      function PrototypeFactory(prototypeSpoon, prototypeFork) {
        this.prototypeSpoon = prototypeSpoon;
        this.prototypeFork = prototypeFork;
      }

      PrototypeFactory.prototype.makeSpoon = function() {
        return this.prototypeSpoon.clone();
      };

      PrototypeFactory.prototype.makeFork = function() {
        return this.prototypeFork.clone();
      };

      return PrototypeFactory;

    })();
    return PrototypeFactory;
  });

}).call(this);

File: python/src/Creational/Prototype/PrototypeFactory.py

#//Prototype Overview
#//Make new objects by cloning the objects which you set as prototypes
#//Still reading? Save your time, watch the video lessons!
#//Video tutorial on design patterns

# PrototypeFactory - a Factory for Prototypes

class PrototypeFactory:

	def __init__(self,prototypeSpoon,prototypeFork):
		self.prototypeSpoon = prototypeSpoon
		self.prototypeFork = prototypeFork
	
	def makeSpoon(self) : 
		return self.prototypeSpoon.clone()

	def makeFork(self) : 
		return self.prototypeFork.clone()
